packages/sanity/src/core/form/inputs/PortableText/object/InlineObject.tsx

```diff
@@ -263,6 +264,11 @@ export const DefaultInlineObjectComponent = (props: BlockProps): React.JSX.Eleme
   const hasWarning = validation.filter((v) => v.level === 'warning').length > 0
   const {enabled: nestedObjectNavigaionEnabled} = useEnhancedObjectDialog()
 
+  // For inline references, always use the normal dialog (not the enhanced dialog)
+  // This allows the reference picker to work properly
+  const isReference = isReferenceSchemaType(schemaType)
+  const shouldUseEnhancedDialog = nestedObjectNavigaionEnabled && !isReference
+
```

Comment:
Made it that a reference that is an inline object in a PTE uses still an older type modal (what it used before) - you can see this example in the referencesinPopover e2e test or in the all the bells and whistles documents by picking a inline reference. The reason behind it is that in order to try to thread this needle within the nested dialog felt like trying to fix something that would potentially be very complex with a simplified way of doing it: if it's a reference, just open it as it used to. It feels it would be a waste of time and potentially unneeded overengineering

<img width="1454" height="474" alt="image" src="https://github.com/user-attachments/assets/db0db744-3406-4dd2-af14-72c05881a9ec" />